ui/ducks/send/send.js

```diff
@@ -705,12 +728,85 @@ const slice = createSlice({
       slice.caseReducers.calculateGasTotal(state);
     },
     /**
-     * sets the provided gasPrice in state and then recomputes the gasTotal
+     * Sets the appropriate gas fees in state and determines and sets the
+     * appropriate transactionType based on gas fee fields received.
      */
-    updateGasPrice: (state, action) => {
-      state.gas.gasPrice = addHexPrefix(action.payload);
+    updateGasFees: (state, action) => {
+      if (
+        action.payload.transactionType === TRANSACTION_ENVELOPE_TYPES.FEE_MARKET
+      ) {
+        state.gas.maxFeePerGas = addHexPrefix(action.payload.maxFeePerGas);
+        state.gas.maxPriorityFeePerGas = addHexPrefix(
+          action.payload.maxPriorityFeePerGas,
+        );
+        state.transactionType = TRANSACTION_ENVELOPE_TYPES.FEE_MARKET;
+      } else {
+        // Until we remove the old UI we don't want to update gasPrice if the
+        // user has changed the field value. When receiving a new estimate the
+        // isAutomaticUpdate property will be on the payload (and set to true).
+        // If isAutomaticUpdate is true, then we check if the previous estimate
+        // was '0x0' or if the previous gasPrice equals the previous
+        // gasEstimate. if either of those cases are true then we update the
+        // gasPrice otherwise we skip it because it indicates the user has
+        // ejected from the estimates by modifying the field.
+        if (
+          action.payload.isAutomaticUpdate !== true ||
```

Comment:
Note that the one case were we can be in this else block and `isAutomaticUpdate` is true is when `updateGasPrice` is called, which is apparently a temporary placeholder to support the old UI